src/zcbor_decode.c

```diff
@@ -349,7 +349,9 @@ static bool str_start_decode(zcbor_state_t *state,
 
 static bool str_overflow_check(zcbor_state_t *state, struct zcbor_string *result)
 {
-	if (result->len > (state->payload_end - state->payload)) {
+	/* Casting to size_t is safe as long as this function is called after
+	 * str_start_decode() which checks that payload_end is bigger that payload. */
```

Comment:
Can't we pull this check out to a helper function? And have the check there as well? I don't like these types of dependencies across code-bases.

Reply:
Good idea. Fixed now.